lib/msal-browser/src/controllers/StandardController.ts

```diff
@@ -2386,6 +2391,41 @@ export class StandardController implements IController {
         );
         return res;
     }
+
+    private logMultipleInstances(
+        performanceEvent: InProgressPerformanceEvent
+    ): void {
+        const clientId = this.config.auth.clientId;
+
+        // @ts-ignore
+        window.msal = window.msal || {};
+        // @ts-ignore
+        window.msal.clientIds = window.msal.clientIds || [];
+
+        // @ts-ignore
+        const clientIds: string[] = window.msal.clientIds;
+
+        for (const currentId of clientIds) {
+            if (currentId === clientId) {
+                this.logger.warning(
+                    "There is already an instance of MSAL.js in the window with the same client id."
+                );
+                // @ts-ignore
+                window.msal.clientIds.push(clientId);
+                collectInstanceStats(clientId, performanceEvent);
+                return;
+            }
+        }
```

Comment:
I think you can cut this whole block and instead log your warning inside `collectInstanceStats` if `sameClientIdInstanceCount > 1`